### PR TITLE
Use block creator for the fairness constraint

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
Use the block creator instead of an intermediate peer for the fairness constraint.

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3904

### CODE REVIEW

**Comment by sebtomba** [comment]
bors try

File: models/src/main/protobuf/CasperMessage.proto [hunk 1]
```diff
@@ -100,7 +100,8 @@ message BlockMessageProto {
 
 message BlockHashMessageProto {
   option (scalapb.message).extends = "CasperMessageProto";
-  bytes  hash       = 1;
+  bytes  hash         = 1;
+  bytes  blockCreator = 2;
```
Comment (zsluedem) [inline models/src/main/protobuf/CasperMessage.proto:104]: Is `blockCreator` the same as `sender`?
Maybe we should use the same name or add a comment here to let others know this is the same as sender.

Comment (sebtomba) [reply models/src/main/protobuf/CasperMessage.proto:104]: @zsluedem IMO we should start fixing the misleading field name `sender`. I'll add a comment

**Review by zsluedem — APPROVED** [review]

**Review by itegulov — APPROVED** [review]

**Pull request merged**